src/helpers/sharingHelper.js

```diff
@@ -485,4 +485,35 @@ module.exports = {
       throw new Error("Invalid Share Type" + lastShare);
     }
   },
+  /**
+   * updates the shared file/folder permission by sharer
+   *
+   * @param shauser
+   * @param resource
+   * @param permissionsreTypeString
+   */
+  updateSharedFilePermissionByUser: async function (
+    sharer,
+    resource,
+    permissions
+  ) {
+    let shareID;
+    const shares = await this.getAllSharesSharedByUser(sharer);
+    for (const shareElement of shares) {
+      if (shareElement.uid_owner === sharer && shareElement.path === resource) {
+        shareID = shareElement.id.toString();
+        break;
+      }
+    }
+    if (!shareID) {
+      return Promise.reject(new Error("Could not get shareId for last share"));
+    }
+    const apiURL = `apps/files_sharing/api/v1/shares/${shareID}`;
+    const body = new URLSearchParams();
+    body.append("permissions", permissions);
+    return httpHelper.putOCS(apiURL, sharer, body.toString()).then((res) => {
+      httpHelper.checkStatus(res, "Could not update share.");
```

Comment:
maybe also report the error code and response body so that the developer can guess why it could not be updated